src/lib/blockchains/nxt.coffee

```diff
@@ -0,0 +1,215 @@
+###
+
+dnschain
+http://dnschain.net
+
+Copyright (c) 2014 okTurtles Foundation
+
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this
+file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+###
+
+module.exports = (dnschain) ->
+    # expose these into our namespace
+    for k of dnschain.globals
+        eval "var #{k} = dnschain.globals.#{k};"
+
+    BlockchainResolver = require('../blockchain.coffee')(dnschain)
+    ResolverStream  = require('../resolver-stream')(dnschain)
+
+    QTYPE_NAME = dns2.consts.QTYPE_TO_NAME
+    NAME_QTYPE = dns2.consts.NAME_TO_QTYPE
+    NAME_RCODE = dns2.consts.NAME_TO_RCODE
+    RCODE_NAME = dns2.consts.RCODE_TO_NAME
+    BLOCKS2SEC = 60
+
+    # Specifications listed here:
+    # - https://wiki.namecoin.info/index.php?title=Welcome
+    # - https://wiki.namecoin.info/index.php?title=Domain_Name_Specification#Importing_and_delegation
+    # - https://wiki.namecoin.info/index.php?title=Category:NEP
+    # - https://wiki.namecoin.info/index.php?title=Namecoin_Specification
+    VALID_NXT_DOMAINS = /^[a-zA-Z0-9]+\/.+/
+
+    class NxtResolver extends BlockchainResolver
+        constructor: (@dnschain) ->
+            @log = gNewLogger 'NXT'
+            @name = 'nxt'
+            @tld = 'nxt'
+
+        config: ->
+            @log.debug "Loading #{@name} resolver"
+
+            params = ["port", "connect"].map (x)-> gConf.get "nxt:#{x}"
+            params[0] ?= "7876"
+            params[1] ?= "127.0.0.1"
+
+            @peer = 'http://' + params[1] + ':' + params[0] + '/nxt?requestType=getAlias&aliasName='
+
+            # TODO: $create doesn't actually connect. you need to open a raw socket
+            #       or an http socket and see if that works before declaring it works
+            @log.info "Nxt API on: %s:%d", params[1], params[0]
+            @
```

Comment:
There should be a fail case in the event that this blockchain doesn't exist on the system that DNSChain is running on.

This method should return `null` or `undefined` if that the blockchain doesn't exist. Show how do you think that should be done here? Perhaps it simply do that if `getConf.get "nxt:..."` doesn't exist?